dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/importers/ui/form/FormItem.java

```diff
@@ -0,0 +1,61 @@
+package edu.harvard.iq.dataverse.importers.ui.form;
+
+import edu.harvard.iq.dataverse.importer.metadata.ImporterData;
+import edu.harvard.iq.dataverse.importer.metadata.ImporterFieldType;
+import edu.harvard.iq.dataverse.importer.metadata.SafeBundleWrapper;
+
+public class FormItem {
+    private final String viewId;
+    private final ImporterData.ImporterField importerField;
+    private final SafeBundleWrapper bundle;
+
+    private Object value;
+
+    // -------------------- CONSTRUCTORS --------------------
+
+    public FormItem(String viewId, ImporterData.ImporterField importerField, SafeBundleWrapper bundle) {
+        this.viewId = viewId;
+        this.importerField = importerField;
+        this.bundle = bundle;
+    }
+
+    // -------------------- GETTERS --------------------
+
+    public String getViewId() {
+        return viewId;
+    }
+
+    public ImporterFieldType getType() {
+        return importerField.fieldType;
+    }
+
+    public String getLabel() {
+        return bundle.getString(importerField.labelKey);
+    }
+
+    public String getDescription() {
+        return bundle.getString(importerField.descriptionKey);
+    }
+
+    public ImporterData.ImporterField getImporterField() {
+        return importerField;
+    }
+
+    public Boolean getRequired() {
+        return importerField.required;
+    }
+
+    public Object getValue() {
```

Comment:
Yes, but this is used on view, so I have to stick to JavaBeans conventions. However I can make second getter for other uses.

Reply:
After a trial I decided to not use `Optional` in this case:
1. Some null-checking that is done with `getValue` will have to be done nevertheless and this will cause unidiomatic code.
2. There is null-checking mechanism for required fields, so if importer will use **required** values it can assume them to be non-null.
3. For the non-required values importer can use Optional internally.
4. This change would also induce the change of signature of input data collection for importer interface, and this will make them less clear.

Reply:
After giving it some thought I think I can agree, since required gives more info and also it is used in view so JSF kinda ruins it.